cyber/common/global_data.cc

```diff
@@ -41,7 +41,7 @@ namespace {
 const std::string& kEmptyString = "";
 std::string program_path() {
   char path[PATH_MAX];
-  auto len = readlink("/proc/self/exe", path, sizeof(path));
+  auto len = readlink("/proc/self/exe", path, sizeof(path) - 1);
```

Comment:
Good catch.

Reply:
How about `boost::filesystem::canonical()`? But it will bind us to boost.